pkg/webhook/webhookreporter.go

```diff
@@ -0,0 +1,88 @@
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/trivy-operator/pkg/operator/etc"
+	"github.com/aquasecurity/trivy-operator/pkg/operator/predicate"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"net/http"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+)
+
+type WebhookReconciler struct {
+	logr.Logger
+	etc.Config
+	client.Client
+}
+
+//+kubebuilder:rbac:groups=aquasecurity.github.io,resources=vulnerabilityreports,verbs=get;list;watch;delete
+
+func (r *WebhookReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	installModePredicate, err := predicate.InstallModePredicate(r.Config)
+	if err != nil {
+		return err
+	}
+
+	reports := []client.Object{
+		&v1alpha1.VulnerabilityReport{},
+		&v1alpha1.ExposedSecretReport{},
+	}
+
+	for _, reportType := range reports {
+		err = ctrl.NewControllerManagedBy(mgr).
+			For(reportType, builder.WithPredicates(
+				predicate.Not(predicate.IsBeingTerminated),
+				installModePredicate)). // TODO: Is there a predicate that determines existance?
```

Comment:
@chen-keinan - is there a predicate that can determine existence? For instance, if I add to the list of supported reports slice, `clustercompliancereport` and let's say even if the report generation is enabled as part of the trivy operator config but hasn't been generated just yet, this is what I see in the logs:

```

1.6625969315044768e+09  ERROR   controller-runtime.source       if kind is a CRD, it should be installed before calling Start   {"kind": "ClusterComplianceDetailReport.aquasecurity.github.io", "error": "no matches for kind \"ClusterComplianceDetailReport\" in version \"aquasecurity.github.io/v1alpha1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1.1
        /Users/simar/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/source.go:139
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext
        /Users/simar/go/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:233
k8s.io/apimachinery/pkg/util/wait.WaitForWithContext
        /Users/simar/go/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:660
k8s.io/apimachinery/pkg/util/wait.poll
        /Users/simar/go/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:594
k8s.io/apimachinery/pkg/util/wait.PollImmediateUntilWithContext
        /Users/simar/go/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:545
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start.func1
        /Users/simar/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/source/source.go:132
```

Reply:
yes , currently the `compliancereports` are disable , ignore these reports for now , we will add it back later